api/v1alpha2/virtualmachine_network_types.go

```diff
@@ -44,6 +44,15 @@ type VirtualMachineNetworkInterfaceSpec struct {
 	// +optional
 	Network common.PartialObjectRef `json:"network,omitempty"`
 
+	// GuestDeviceName is used to rename the device inside the guest when the
+	// bootstrap provider is Cloud-Init. Please note it is up to the user to
+	// ensure the provided device name does not conflict with any other devices
+	// inside the guest, ex. dvd, cdrom, sda, etc.
+	//
+	// +optional
+	// +kubebuilder:validation:Pattern=^\w\w+$
```

Comment:
Do we wanna match the pattern in Name? `Pattern="^[a-z0-9]{2,}$"` will exclude underscores.

Reply:
The validation for this field is more lenient since we don't use it for the k8s CR name for the network inteface.